otherlibs/fiber/pool.ml

```diff
@@ -1,17 +1,13 @@
 open Stdune
 open Core
-open Core.O
-
-type mvar =
-  | Done
-  | Task of (unit -> unit t)
 
 type status =
   | Open
   | Closed
 
-type t =
-  { mvar : mvar Mvar.t
+type nonrec t =
+  { tasks : (unit -> unit t) Queue.t
+  ; mutable reader : unit k option
```

Comment:
Is `reader` a "runner"? I'm not entirely sure how to interpret this field.

Reply:
Indeed runner would be a better name. It is the continuation on which `run` is waiting on.